Nodejs/Product/Nodejs/TypingsAcquisition.cs

```diff
@@ -0,0 +1,143 @@
+            string pathToRootNpmDirectory,
+            string pathToRootProjectDirectory,
+            IEnumerable<IPackage> packages,
+            Redirector redirector) {
+            if (!packages.Any()) {
+                return true;
+            }
+
+            var tsdPath = GetTsdPath(pathToRootNpmDirectory);
+            if (tsdPath == null) {
+                if (redirector != null) {
+                    redirector.WriteErrorLine(SR.GetString(SR.TsdNotInstalledError));
+                }
+                return false;
+            }
+
+            using (var process = ProcessOutput.Run(
+                tsdPath,
+                TsdInstallArguments(packages),
+                pathToRootProjectDirectory,
+                null,
+                false,
+                redirector,
+                quoteArgs: true)) {
+                if (!process.IsStarted) {
+                    // Process failed to start, and any exception message has
+                    // already been sent through the redirector
+                    if (redirector != null) {
+                        redirector.WriteErrorLine("could not start tsd");
+                    }
+                    return false;
+                }
+                var i = await process;
+                if (i == 0) {
+                    if (redirector != null) {
+                        redirector.WriteLine(SR.GetString(SR.TsdInstallCompleted));
+                    }
+                    return true;
+                } else {
+                    process.Kill();
+                    if (redirector != null) {
+                        redirector.WriteErrorLine(SR.GetString(SR.TsdInstallErrorOccurred));
+                    }
+                    return false;
+                }
+            }
+        }
+
+        private static IEnumerable<string> TsdInstallArguments(IEnumerable<IPackage> packages) {
+            return new[] { "install", }.Concat(packages.Select(GetPackageTsdName)).Concat(new[] { "--save" });
+        }
+
+        private static IEnumerable<string> CurrentTypingsPackages(string pathToRootProjectDirectory) {
+            var packages = new List<string>();
+            var typingsDirectoryPath = Path.Combine(pathToRootProjectDirectory, TypingsDirectoryName);
+            if (!Directory.Exists(typingsDirectoryPath)) {
+                return packages;
+            }
+            try {
+                foreach (var file in Directory.EnumerateFiles(typingsDirectoryPath, "*.d.ts", SearchOption.AllDirectories)) {
+                    var directory = Directory.GetParent(file);
+                    if (directory.FullName != typingsDirectoryPath && Path.GetFullPath(directory.FullName).StartsWith(typingsDirectoryPath)) {
+                        packages.Add(directory.Name);
+                    }
+                }
+            } catch (SystemException e) when (e is IOException || e is SecurityException || e is UnauthorizedAccessException) {
```

Comment:
man, already making breaking changes to VS2013! :wink: For the sake of our CI... let's at least wait until after we have a conditional AppVeyor build configuration before we start using all the fancy new features. :smiley: